File: hw1/hw1.py
#rsait001 Ryota Saito 
# writing "cat" cmd in python 

#imports
import sys

#pop off hw1.py from the cmd line
sys.argv.pop(0)

#check for 0 args (1 is the base command) 
if len(sys.argv) > 0:

	#parse files (i returns files name)
 	for i in sys.argv:

		#open file, read it and print it
		file = open(i, 'r')
		print "---------------------------------------------------------------------------------------------------------------------------------------------"
		print i 
		print file.read()

else:
	print "Must supply file(s)" 

File: hw1/hw1.pl
#!/usr/bin/perl
#rsait001 Ryota Saito 
# writing "cat" cmd in perl

#@ARGV stores cmdline args into an array
if(scalar @ARGV == 0){
	exit 0
}

foreach $i (@ARGV){
	# $i now contains all text file names
	#open file for reading (< means stdin, $fh is file descriptor)
	open( $fh, '<', $i);
	#$line i in the scope of the file
	while ( $line = <$fh> ) {
		print "$line";
	}

    close $fh;
} 
